imblearn/over_sampling/_smote/filter.py

```diff
@@ -345,10 +344,8 @@ def _validate_estimator(self):
 
         if self.svm_estimator is None:
             self.svm_estimator_ = SVC(gamma="scale", random_state=self.random_state)
-        elif isinstance(self.svm_estimator, SVC):
-            self.svm_estimator_ = clone(self.svm_estimator)
         else:
-            raise_isinstance_error("svm_estimator", [SVC], self.svm_estimator)
+            self.svm_estimator_ = clone(self.svm_estimator)
```

Comment:
This change enables SVM estimators that enforce the same API contract as sklearn to be integrated instead.  This would enable SVMSMOTE to utilize GPU-accelerated estimators from cuML, providing significant performance gains when working with large datasets.

![image](https://user-images.githubusercontent.com/86264103/133164608-c1014bb0-ec5d-4f50-8b2f-528e85e4819d.png)

Hardware Specs for the Loose Benchmark:
Intel Xeon E5-2698, 2.2 GHz, 16-cores & NVIDIA V100 32 GB GPU

Benchmarking gist:
https://gist.github.com/NV-jpt/039a8d9c7d37365379faa1d7c7aafc5e